### ISSUE TITLE
Simplify adding frontend routes in a barebone application

### ISSUE THREAD

**Issue by platosha** [vaadin/hilla#920]
As a developer, I want to start developing a Hilla application from a clean Spring Boot based template. I use a Spring Boot starter, then add all necessary Hilla dependencies, and then start the application in dev mode for the first time. The application starts, but as a next step, before adding my first view, I should create the following manually:

- the `frontend` directory
- `index.html`
- `index.ts`
- `App.tsx`
- `routes.tsx`

It would be great to have a tool that generates these for me after running some command.

In addition, having some instructions pointing into this tool would be great in the log messages during the first run, in the `HELP.md` file, and in Hilla docs.

**Comment by marcushellberg** [comment]
Please consider [generating an `index.tsx` instead](https://github.com/vaadin/start/issues/2198#issuecomment-1421006314), keeping the structure in line with "standard" React convention used by Create React App https://create-react-app.dev/docs/getting-started

**Comment by tarekoraby** [comment]
@marcushellberg, let's deal with the `index.tsx` enhancement request separately. It will be weird for the CLI to generate one set of files for a completely new project, and another for a project without a frontend folder.

**Comment by marcushellberg** [comment]
Why can't we make both consistent by using index.tsx?

**Comment by tarekoraby** [comment]
Sure, if that's as simple as changing the extension in a couple of places, then let's do it. If it will take some non-trivial amount of effort, then let's prioritize it separately in the backlog.

**Comment by marcushellberg** [comment]
It should be just that, the support for `index.tsx` was added on the backend already a few months ago. I've been making this change in my projects locally since then to keep them more aligned with React practices.

**Comment by platosha** [comment]
Yes, `index.tsx` would make sense right away, since it's supported AFAIK.

**Comment by platosha** [comment]
@marcushellberg proposed (and @cromoteca prototyped) adding a plugin goal for this feature. I think, it makes sense as a plugin goal, as we want to help users with Java background here.
